framework/src/main/java/org/checkerframework/common/accumulation/AccumulationTransfer.java

```diff
@@ -0,0 +1,140 @@
+package org.checkerframework.common.accumulation;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import java.util.Arrays;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The default transfer function for an accumulation checker.
+ *
+ * <p>Subclasses should call the {@link #accumulate(Node, TransferResult, String...)} accumulate}
+ * method to accumulate a string at a particular program point.
+ */
+public class AccumulationTransfer extends CFTransfer {
+
+    /** The type factory. */
+    protected final AccumulationAnnotatedTypeFactory typeFactory;
+
+    /**
+     * Build a new AccumulationTransfer for the given analysis.
+     *
+     * @param analysis the analysis
+     */
+    public AccumulationTransfer(CFAnalysis analysis) {
+        super(analysis);
+        typeFactory = (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Updates the estimate of how many things {@code node} has accumulated.
+     *
+     * <p>If the node is an invocation of a method that returns its receiver, then its receiver's
+     * type will also be updated. This chain will continue as long as each receiver is itself a
```

Comment:
Are there tests of this functionality?  Test `SimpleFluent.java` does not seem to test it.  I expect a test would look like
```
  s.a().b().c();
  s.build();
```